### PR TITLE
REALMC-8543: Implement the schema models command

### PR DESCRIPTION
**Pull request by makesitgo**

### CODE REVIEW

File: internal/cloud/realm/schema.go [hunk 1]
```diff
@@ -1,5 +1,60 @@
...
+	Imports   []string           `json:"import_statements"`
+	Code      string             `json:"schema"`
+	Warnings  []SchemaModelAlert `json:"warnings"`
+	Error     SchemaModelAlert   `json:"error"`
+}
+
+type SchemaModelAlert struct {
+	Message string `json:"error"`
+	Code    string `json:"error_code"`
+}
```
Comment (nathantfrank) [inline internal/cloud/realm/schema.go:39]: [q] Understand this is coming from the API so nothing to be done here, but maybe something to keep in mind later. Does it make sense that a `Warning`'s alert is considered an `error` vs `message`?

File: internal/commands/schema/models_inputs.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+			i.NoImports = noImports
+		}
+
+		if !i.Flat {
+			var flat bool
+			if err := ui.AskOne(&flat, &survey.Confirm{Message: "Would you like group all generated data models together?"}); err != nil {
+				return err
+			}
+			i.Flat = flat
+		}
```
Comment (nathantfrank) [inline internal/commands/schema/models_inputs.go:70]: [q] Going the route of always asking if flag isn't set?

**Review by nathantfrank — COMMENTED** [review]
Looking good so far, just some Q's and probably should put something for the help text

File: internal/cloud/realm/schema.go (hunk 1 above)
Comment (makesitgo) [reply internal/cloud/realm/schema.go:39]: I'm with you 💯  on this ... I also don't love we use "error code" terminology for a warning, but I think these warnings/errors are considered more supplemental than anything.  this is all simply reverse-engineering & mirroring what the UI does with these things

File: internal/commands/schema/models_inputs.go (hunk 2 above)
Comment (makesitgo) [reply internal/commands/schema/models_inputs.go:70]: haha yeah, based off the feedback from our past bug bash, it felt like users appreciated more prompts/less flags.  fwiw, this "prompt if not set" behavior still will only occur if the user omits `--language` flag ... if they were to set `--language` but not set `--flat` or `--no-imports`, the prompts would still not pop up

File: internal/commands/commands.go [hunk 3]
```diff
@@ -274,4 +275,20 @@ your logs and follow any newly created logs in real-time.`,
...
+		Aliases:     []string{"schemas"},
+		Description: "Manage the schemas of your Realm app",
+		SubCommands: []cli.CommandDefinition{
+			{
+				Command:     &schema.CommandModels{},
+				Use:         "models",
+				Aliases:     []string{"model"},
+				Display:     "schema models",
+				Description: "Generate source code for the schema data models",
+				Help:        `TODO: this`,
```
Comment (UnicodeSnowman) [inline internal/commands/commands.go:290]: What's the plan for this TODO?

**Review by UnicodeSnowman — APPROVED** [review]

File: internal/commands/commands.go (hunk 3 above)
Comment (makesitgo) [reply internal/commands/commands.go:290]: was going to reach out to @kpham1231 for this one

**Pull request merged**